internal/robot/cache.go

```diff
@@ -21,7 +20,7 @@ type cacheRobotClient struct {
 	serversByID map[int]*hrobotmodels.Server
 }
 
-func NewClient(robotClient hrobot.RobotClient, cacheTimeout time.Duration) Client {
+func WithCache(cacheTimeout time.Duration, robotClient Client) Client {
```

Comment:
```suggestion
func NewCachedClient(robotClient Client, cacheTimeout time.Duration) Client {
```

Reply:
I put the client last on purpose. I found it pretty confusing to have the client first when nesting, as the argument `RateLimitWaitTime` is not next to `WithRateLimit` anymore. For comparison:

Client last (current version):

```go
robotClient = robot.WithRateLimit(
  robot.WithCache(
    cfg.Robot.CacheTimeout,
    c
  ),
  cfg.Robot.RateLimitWaitTime,
)
```

Client first (your suggestion):

```go
robotClient = robot.WithRateLimit(
  cfg.Robot.RateLimitWaitTime,
  robot.WithCache(
    cfg.Robot.CacheTimeout,
    c
  ),
)
```

Reply:
Yea, I updated the naming :) Just did not agree with the changed order of arguments in your suggested replacement. (Sorry, should have been more explicit.)